### ISSUE TITLE
amp-sidebar: autoscroll only affects visual scroll position

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31167]
## WCAG Level
Level A

## Priority
Low

## Type
Navigation

## Pages/screens/Components affected
* [amp-sidebar](https://amp.dev/documentation/components/amp-sidebar/)

## Description
For people who navigate using a keyboard or keyboard-like device (such as a switch control or a joystick) the focus order of interactive controls in a document should be logical and intuitive.

When a page changes dynamically as a result of user interaction, it's important that focus is, where necessary, moved to the most appropriate place within the page.

`amp-sidebar` has an `autoscroll` attribute. When the sidebar is displayed, the browser automatically scrolls to the first element in the sidebar with this attribute. While this works visually, keyboard focus is not moved to this element.

## User impact
Sighted keyboard users will see the sidebar scrolled to the relevant element, but once they begin navigating (using `Tab` / `Shift` + `Tab`), they will begin navigation from the very start of the sidebar / from the first focusable element (where the focus is currently placed when the sidebar is opened).

Assistive technology users won't have any indication that the sidebar was visually scrolled to a particular element at all. If a particular element was autoscrolled to highlight it / make it immediately obvious to users, this information will not be immediately conveyed to these users. (Furthermore, note that the current examples/demos for autoscroll also only add a CSS class to visually highlight the autoscrolled element, and do not programmatically convey this highlighting).

## Required solution
If it's important to highlight and scroll to a particular element, then move the initial focus to this element as well when the sidebar is opened/displayed.

Focus is moved programmatically using the standard `.focus()` JavaScript method. This works for elements that are focusable by default (links, buttons, form controls). To set focus to a static element, which does not ordinarily receive focus, you will also need to add `tabindex="-1"` to the element, which signals to the browser that the element can receive programmatic focus.

We'd recommend adding the `tabindex` attribute dynamically, as part of the sidebar/autoscroll scripting (rather than expecting authors to add this themselves).

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied:
1. Turn on your screen reader.
2. Navigate to, and trigger, the sidebar that contains an `autoscroll` element.
3. Verify that not only the visual scroll position of the sidebar was moved, but that focus was also placed on the autoscrolled element.

## Related standards
* [WCAG 2.1 Success Criterion 2.4.3 Focus Order (Level A)](https://www.w3.org/TR/WCAG21/#focus-order)

## More information
* [W3C - Placing the interactive elements in an order that follows sequences and relationships within the content](https://www.w3.org/WAI/WCAG21/Techniques/general/G59)

## Test data
Test date: November 2020
